### ISSUE TITLE
[CI] Failure in PercolateQueryBuilderTests testToQuery

### ISSUE THREAD

**Issue by droberts195** [elastic/elasticsearch#26736]
This test failed in the following build https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+oracle-periodic/668/console

The error is:

```
07:34:06 ERROR   0.40s J2 | PercolateQueryBuilderTests.testToQuery <<< FAILURES!
07:34:06    > Throwable #1: MapperParsingException[failed to parse]; nested: IllegalArgumentException[object field starting or ending with a [.] makes object resolution ambiguous: [ ]];
07:34:06    > 	at __randomizedtesting.SeedInfo.seed([7E7ABCF12BBA9DE:F01CA9F163386C34]:0)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.wrapInMapperParsingException(DocumentParser.java:176)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:69)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:277)
07:34:06    > 	at org.elasticsearch.percolator.PercolateQueryBuilder.doToQuery(PercolateQueryBuilder.java:410)
07:34:06    > 	at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:96)
07:34:06    > 	at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:571)
07:34:06    > 	at java.lang.Thread.run(Thread.java:748)
07:34:06    > Caused by: java.lang.IllegalArgumentException: object field starting or ending with a [.] makes object resolution ambiguous: [ ]
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.splitAndValidatePath(DocumentParser.java:183)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.getMapper(DocumentParser.java:938)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseValue(DocumentParser.java:589)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:396)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:373)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:465)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObject(DocumentParser.java:504)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:383)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:373)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.internalParseDocument(DocumentParser.java:93)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:66)
07:34:06    > 	... 41 more
```

The REPRO command is:

```
gradle :modules:percolator:test \
  -Dtests.seed=7E7ABCF12BBA9DE \
  -Dtests.class=org.elasticsearch.percolator.PercolateQueryBuilderTests \
  -Dtests.method="testToQuery" \
  -Dtests.security.manager=true \
  -Dtests.locale=es-MX \
  -Dtests.timezone=America/Costa_Rica
```

This reproduces locally for me.

**Comment by cbuescher** [comment]
@martijnvg I'm looking into it, its a weird field name as far as I can see. Will have a PR soon.

**Comment by cbuescher** [comment]
The is an offending field name is a whitespace (more specifically a "MEDIUM MATHEMATICAL SPACE", but I guess any whitespace would trigger this). I'll exclude this from the set of options when creating the test document source.

**Comment by cbuescher** [comment]
@martijnvg you already seem to have fixed this with 99d79d5a0f9191cba81ee1288c3fa10de688ba84 on master, I will just cherry-pick this back to 5.6.

**Comment by cbuescher** [comment]
Closed by https://github.com/elastic/elasticsearch/pull/26740
